discovery-provider/src/queries/download_csv.py

```diff
@@ -0,0 +1,205 @@
+from typing import TypedDict
+
+from sqlalchemy import desc, or_
+
+from src.models.playlists.playlist import Playlist
+from src.models.playlists.playlist_route import PlaylistRoute
+from src.models.tracks.track import Track
+from src.models.tracks.track_route import TrackRoute
+from src.models.users.usdc_purchase import PurchaseType, USDCPurchase
+from src.models.users.usdc_transactions_history import (
+    USDCTransactionMethod,
+    USDCTransactionsHistory,
+    USDCTransactionType,
+)
+from src.models.users.user import User
+from src.models.users.user_bank import USDCUserBankAccount
+from src.utils.config import shared_config
+from src.utils.csv_writer import write_csv
+from src.utils.db_session import get_db_read_replica
+
+env = shared_config["discprov"]["env"]
+
+
+class DownloadPurchasesArgs(TypedDict):
+    buyer_user_id: int
+
+
+class DownloadSalesArgs(TypedDict):
+    seller_user_id: int
+
+
+class DownloadWithdrawalsArgs(TypedDict):
+    user_id: int
+
+
+# Get all purchases or sales for a given artist.
+def get_purchases_or_sales(user_id: int, is_purchases: bool):
```

Comment:
Just curious, why not use existing `get_usdc_purchases` query?

Reply:
spent a bit of time looking into consolidating, but I think the added complexity is not worth it.
yes, we would get one source of truth for the query, but we'd end up with an even longer and harder-to-understand function, that would make it scarier to change in the future (i think `get_usdc_purchases` is already complex enough as-is).

will keep separate if you don't feel strongly

Reply:
I'm ok with keeping it separate for now so we can keep moving on other tasks, but I think we will want some insurance that the two tables are the same and remain the same. We can add a new task in the cleanup milestone to add some tests to ensure that if you think that's easier than consolidating

Reply:
added https://linear.app/audius/issue/PAY-1870/add-tests-for-csv-download-queries